### ISSUE TITLE
Improper @covers annotation makes PHPUnit crash with a fatal error

### ISSUE THREAD

**Issue by morozov** [sebastianbergmann/phpunit#4199]
| Q                   | A
| --------------------| ---------------
| PHPUnit version     | 9.1.3
| PHP version         | 7.4.5
| Installation Method | Composer

#### Summary

Although this issue is very similar to #4196, I'm filing it as a separate one since the side effects are slightly different.

If a method is annotated with a `@covers` annotation that points to an existing user-defined class but a non-existing method, PHPUnit crashes with a fatal error.

#### Current behavior

```
$ phpunit --coverage-php coverage.php SomeTest.php

PHPUnit 9.1.3 by Sebastian Bergmann and contributors.

Fatal error: Uncaught ReflectionException: Method SomeTest::whatever() does not exist in vendor/sebastian/code-unit/src/CodeUnit.php:408
Stack trace:
#0 vendor/sebastian/code-unit/src/CodeUnit.php(408): ReflectionMethod->__construct('SomeTest', 'whatever')
#1 vendor/sebastian/code-unit/src/CodeUnit.php(65): SebastianBergmann\CodeUnit\CodeUnit::reflectorForClassMethod('SomeTest', 'whatever')
#2 vendor/sebastian/code-unit/src/Mapper.php(80): SebastianBergmann\CodeUnit\CodeUnit::forClassMethod('SomeTest', 'whatever')
#3 vendor/phpunit/phpunit/src/Util/Test.php(591): SebastianBergmann\CodeUnit\Mapper->stringToCodeUnits('SomeTest::whate...')
#4 vendor/phpunit/phpunit/src/Util/Test.php(97): PHPUnit\Util\Test::getLinesToBeCoveredOrUsed('SomeTest', 'invoke', 'covers')
#5 vendor/phpunit/phpunit/src/Framework/TestResult.php(783): PHPUnit\Util\Tes in vendor/sebastian/code-unit/src/CodeUnit.php on line 411
```

#### How to reproduce

```php
<?php declare(strict_types=1);

use PHPUnit\Framework\TestCase;

class SomeTest extends TestCase
{
    /**
     * @test
     * @covers SomeTest::whatever
     */
    public function invoke() : void
    {
        self::assertTrue(true);
    }
}
```

#### Expected behavior

PHPUnit emits a warning containing the class/method information and ignores the code coverage produced by the test with the invalid annotation but doesn't crash.

**Comment by sebastianbergmann** [comment]
I slightly changed the example code to
```php
<?php declare(strict_types=1);
use PHPUnit\Framework\TestCase;

final class Test extends TestCase
{
    /**
     * @covers SomeTest::whatever
     */
    public function testOne(): void
    {
        self::assertTrue(true);
    }
}
```

and put it into `tests/Test.php`. I created an `src` directory next to `tests` with an empty PHP file:
```
.
├── src
│   └── empty.php
└── tests
    └── Test.php

2 directories, 2 files
```

When I run this then I get
```
$ phpunit --coverage-clover /tmp/clover.xml --whitelist src tests
PHPUnit 9.1.3-4-gd5cd186cb by Sebastian Bergmann and contributors.

"@covers SomeTest::whatever" is invalid
```

When I change the annotation to `@covers Foo::bar` then I get
```
$ phpunit --coverage-clover /tmp/clover.xml --whitelist src tests
PHPUnit 9.1.3-4-gd5cd186cb by Sebastian Bergmann and contributors.

Fatal error: Uncaught ReflectionException: Method Foo::bar() does not exist in /usr/local/src/phpunit/vendor/sebastian/code-unit/src/CodeUnit.php:408
Stack trace:
#0 /usr/local/src/phpunit/vendor/sebastian/code-unit/src/CodeUnit.php(408): ReflectionMethod->__construct('Foo', 'bar')
#1 /usr/local/src/phpunit/vendor/sebastian/code-unit/src/CodeUnit.php(65): SebastianBergmann\CodeUnit\CodeUnit::reflectorForClassMethod('Foo', 'bar')
#2 /usr/local/src/phpunit/vendor/sebastian/code-unit/src/Mapper.php(80): SebastianBergmann\CodeUnit\CodeUnit::forClassMethod('Foo', 'bar')
#3 /usr/local/src/phpunit/src/Util/Test.php(591): SebastianBergmann\CodeUnit\Mapper->stringToCodeUnits('Foo::bar')
#4 /usr/local/src/phpunit/src/Util/Test.php(97): PHPUnit\Util\Test::getLinesToBeCoveredOrUsed('Test', 'testOne', 'covers')
#5 /usr/local/src/phpunit/src/Framework/TestResult.php(783): PHPUnit\Util\Test::getLinesToBeCovered('Test', 'testOne')
#6 /usr/local/src/phpunit/src/Framework/TestCase.php(772): PHPUnit\Framework\TestResult->run(Object(Test))
#7 /usr/local/src/phpunit/src/Framework/TestSuite.php(639): PHPUnit\Framework\TestCase->run(Object(PHPUnit\Framework\TestResult))
#8 /usr/local/src/phpunit/src/Framework/TestSuite.php(639): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#9 /usr/local/src/phpunit/src/TextUI/TestRunner.php(653): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#10 /usr/local/src/phpunit/src/TextUI/Command.php(108): PHPUnit\TextUI\TestRunner->run(Object(PHPUnit\Framework\TestSuite), Array, Array, true)
#11 /usr/local/src/phpunit/src/TextUI/Command.php(68): PHPUnit\TextUI\Command->run(Array, true)
#12 /usr/local/src/phpunit/phpunit(61): PHPUnit\TextUI\Command::main()
#13 {main}

Next SebastianBergmann\CodeUnit\ReflectionException: Method Foo::bar() does not exist in /usr/local/src/phpunit/vendor/sebastian/code-unit/src/CodeUnit.php:411
Stack trace:
#0 /usr/local/src/phpunit/vendor/sebastian/code-unit/src/CodeUnit.php(65): SebastianBergmann\CodeUnit\CodeUnit::reflectorForClassMethod('Foo', 'bar')
#1 /usr/local/src/phpunit/vendor/sebastian/code-unit/src/Mapper.php(80): SebastianBergmann\CodeUnit\CodeUnit::forClassMethod('Foo', 'bar')
#2 /usr/local/src/phpunit/src/Util/Test.php(591): SebastianBergmann\CodeUnit\Mapper->stringToCodeUnits('Foo::bar')
#3 /usr/local/src/phpunit/src/Util/Test.php(97): PHPUnit\Util\Test::getLinesToBeCoveredOrUsed('Test', 'testOne', 'covers')
#4 /usr/local/src/phpunit/src/Framework/TestResult.php(783): PHPUnit\Util\Test::getLinesToBeCovered('Test', 'testOne')
#5 /usr/local/src/phpunit/src/Framework/TestCase.php(772): PHPUnit\Framework\TestResult->run(Object(Test))
#6 /usr/local/src/phpunit/src/Framework/TestSuite.php(639): PHPUnit\Framework\TestCase->run(Object(PHPUnit\Framework\TestResult))
#7 /usr/local/src/phpunit/src/Framework/TestSuite.php(639): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#8 /usr/local/src/phpunit/src/TextUI/TestRunner.php(653): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#9 /usr/local/src/phpunit/src/TextUI/Command.php(108): PHPUnit\TextUI\TestRunner->run(Object(PHPUnit\Framework\TestSuite), Array, Array, true)
#10 /usr/local/src/phpunit/src/TextUI/Command.php(68): PHPUnit\TextUI\Command->run(Array, true)
#11 /usr/local/src/phpunit/phpunit(61): PHPUnit\TextUI\Command::main()
#12 {main}
  thrown in /usr/local/src/phpunit/vendor/sebastian/code-unit/src/CodeUnit.php on line 411
```

Weird. I'll have a look! And thank you for bringing this to my attention.

**Comment by sebastianbergmann** [comment]
This issue is solved by `sebastian/code-unit` 1.0.2. Next stop: #4196.

**Comment by morozov** [comment]
Thank you, @sebastianbergmann. PHPUnit works as expected both in this case and #4196.
